cmd/controller/main.go

```diff
@@ -84,6 +84,7 @@ const (
 	kubeconfigFlag               = "kubeconfig"
 	allocationBatchWaitTime      = "allocation-batch-wait-time"
 	defaultResync                = 30 * time.Second
+	podNamespace                 = "pod-namespace"
```

Comment:
I would just call this `lease-namespace` and make it clear it's the namespace to use for the controller lease. The helm chart can make sure it's aligned appropriately.

Strictly speaking, I don't think this needs to be in the same namespace as the pod - I think the controller just needs rights to it and for it to exist, which was the issue, AIUI.

Reply:
Would this be confusing since the namespace is the same one as the controller pods? Or should there be another env variable for leader election?

Reply:
As a note, this doesn't (and probably shouldn't?) come from helm. We can use the downward API to get this information from K8s directly:

https://kubernetes.io/docs/concepts/workloads/pods/downward-api/

Reply:
Just thinking, someone could do a yaml install to a different namespace, and expect it to work. If we use the downward API, this handles that scenario.

Reply:
It should be already using the downward api: https://github.com/googleforgames/agones/blob/main/install/helm/agones/templates/controller.yaml#L159-L162 is there a big difference between exposing it as a env variable vs a downward-api volume?

Reply:
Oh sweet - I didn't realise it was already there.

If it's already there as an env var, then you are good to go!

Reply:
If this is already plumbed into the deployment then I have no problem with the existing name - AIUI you're basically consuming an env variable that we already set here, so I guess my `lease-namespace` comment is ignorable.